File: android/src/main/java/ch/nevis/mobile/authentication/sdk/react/methods/LocalAuthenticatorsMethodHandler.kt
/*
 * Copyright © 2023 Nevis Security AG. All rights reserved.
 */

package ch.nevis.mobile.authentication.sdk.react.methods

import ch.nevis.mobile.authentication.sdk.react.MethodChannelHandler
import ch.nevis.mobile.authentication.sdk.react.model.ReactMethodName
import ch.nevis.mobile.authentication.sdk.react.model.message.incoming.ChannelInMessage
import ch.nevis.mobile.authentication.sdk.react.model.message.outgoing.LocalAuthenticatorsOutMessage
import ch.nevis.mobile.authentication.sdk.react.model.sdk.toDto
import ch.nevis.mobile.sdk.api.MobileAuthenticationClient

class LocalAuthenticatorsMethodHandler : MethodHandler() {
    override fun execute(client: MobileAuthenticationClient, channelInMessage: ChannelInMessage) {
        val resultMessage = LocalAuthenticatorsOutMessage(
            channelInMessage.operationId,
            client.localData().authenticators().map { it.toDto(null) }.toList()
        )

        MethodChannelHandler.resolve(ReactMethodName.LOCAL_AUTHENTICATORS, resultMessage)
    }
}

File: ios/Methods/LocalAuthenticatorsMethodHandler.swift
//
// Copyright © 2023 Nevis Security AG. All rights reserved.
//

import NevisMobileAuthentication

struct LocalAuthenticatorsMethodHandler: MethodHandler {
	func execute(using client: MobileAuthenticationClient, with message: ChannelInMessage) throws {
		let resultMessage = LocalAuthenticatorsOutMessage(operationId: message.operationId,
															 authenticators: client.localData.authenticators)
		MethodChannelHandler.shared.resolve(method: .localAuthenticators, message: resultMessage)
	}
}
